scrapperFunctions/index.js

```diff
@@ -0,0 +1,227 @@
+  }
+};
+
+/**
+ * Collect the data of different screens
+ * @param {string} tableSelector The identifier of the table
+ * @param {string} rowSelector How can I find the number of rows in the table
+ * @param {string} nextPageButton The identifier of the button
+ * @param {int | double} time How much time wait for do the process again
+ * @param {function} callback A function needed in progressBar
+ * @returns Array[Array[string]]
+ */
+// eslint-disable-next-line space-before-function-paren
+ScrapPage.prototype.checkData = async function (
+  tableSelector,
+  rowSelector,
+  nextPageButton,
+  time,
+  // eslint-disable-next-line prettier/prettier
+  callback,
+) {
+  try {
+    let exp = await this.expectedRows(rowSelector);
+    let data = await this.getDataTable(tableSelector);
+    const validateNumber = /^[0-9]*$/;
+    const onlyNumbers = validateNumber.test(exp);
+    typeof exp !== 'string' && onlyNumbers ? (exp = data.length) : exp;
+    let obt = data.length;
+    if (exp == 0) return exp;
+    console.log(`Expected data: ${exp}`);
+    console.log(`Getting data...`);
+
+    while (exp > obt) {
+      await this.progressBar(data.length, exp, callback);
+      await this.clickButton(nextPageButton, time);
+      const newData = await this.getDataTable(tableSelector);
+      data = data.concat(newData);
+      obt += newData.length;
+    }
+    await this.progressBar(data.length, exp, callback);
+    return data;
+  } catch (err) {
+    console.error(`Error: ${err}`);
+    throw err;
+  }
+};
+
+/**
+ * A function to know the percentage of data obtained
+ * @param {int} data The quantity of data obtained at this moment
+ * @param {int} expected How much data should be
+ * @param {function} callback How print thw data
+ */
+// eslint-disable-next-line space-before-function-paren
+ScrapPage.prototype.progressBar = function (data, expected, callback) {
+  try {
+    const percentage = Math.round((100 * data) / expected);
+    callback(percentage);
+    return;
+  } catch (err) {
+    console.error(`Error: ${err}`);
+    throw err;
+  }
+};
+
+/**
+ * Obtain the data from a table
+ * @param {string} select How find the table
+ * @returns [Array[Array[string]]]
+ */
+// eslint-disable-next-line space-before-function-paren
+ScrapPage.prototype.getDataTable = async function (select) {
+  try {
+    const data = await this.page.evaluate(tableToArrays, select);
+    data.shift(); // Delete column headings
+    return data;
+  } catch (err) {
+    console.error(`Error: ${err}`);
+    throw err;
+  }
+};
+
+/**
+ * Save the data in a file
+ * @param {*} data Data to save
+ * @param {string} route Where save the data
+ */
+// eslint-disable-next-line space-before-function-paren
+ScrapPage.prototype.saveFile = function (data, route) {
+  const fs = require('fs');
+  fs.writeFile(route, JSON.stringify(data), (error) => {
+    if (error) {
+      console.error(`Error: ${error}`);
+      throw error;
+    } else {
+      return;
+    }
+  });
```

Comment:
@wolftrax5, as a follow up, you should avoid any writing to the server with the bot, maybe you can setup a redis db with a cache no more than 3 days for this